decidim-surveys/spec/queries/decidim/surveys/survey_user_answers_spec.rb

```diff
@@ -14,6 +14,7 @@
 
   it "returns the user answers for each user" do
     result = subject.query
+    result.reverse! unless result.first.first.user == user_1 # ensures that the first record is the first user
 
     expect(result[0]).to eq(survey_answers_user_1.sort { |answer| answer.question.position })
```

Comment:
@leio10 For these things, I usually do something like `expect(result).to contain_exactly(survey_answers_user_1, survey_answers_user_2)`. Would that work here?

Reply:
I'm not totally sure. I don't like it either but, according to the `sort` calls below, I guess that the answers could also be retrieved with a different order each time. 

Maybe we can do this
```ruby
result.reverse! unless result.first.first.user == user_1
expect(result[0]).to contain_exactly(survey_answers_user_1)
expect(result[1]).to contain_exactly(survey_answers_user_2)
```
but I can't figure out a simpler solution.

Reply:
But I think that's what `contain_exactly` do, treat the array as a set so that order doesn't matter...

Reply:
Mmm, and it would work with an array of arrays? 
In other words, this would work?
```ruby
expect([[3, 2, 1], [6, 5, 4]]).to contain_exactly([1, 2, 3], [4, 5, 6])
```
My bet is no, because `contain_exactly` doesn't require that received params are arrays. Anyway, I will check it now... :smiley:

Reply:
And if you keep the ordering that's already there?

```
expect(result).to contain_exactly(survey_answers_user_1.sort { |answer| answer.question.position }, survey_answers_user_2.sort { |answer| answer.question.position })
```

 The query seems to order the answers by position...

Reply:
You are right, I think that should work always :+1:

Reply:
Ther's also `match_array`, but I'm not sure it fits what you need